### ISSUE TITLE
Minju: Headless CMS, 3 weeks

### ISSUE THREAD

**Issue by minjupgeorge** [lab-brussels-1/home#413]
# headless cms
## Learning goals

- 🥚 Understanding what a content management system is, and why people use it 
- 🥚 Given an existing CMS installation fill in and publish a new piece of content
- 🥚 Perform a simple http call via postman to retrieve a piece of information.
- 🐣 Extend and adapt an existing CMS configuration
- 🐣 Perform a http call with filtering via Postman
- 🐣 Perform a http call with "include" via Postman
- 🐣 Perform a http POST call to create new data
- 🐣 Perform a http PUT call to update existing data
- 🐣 Configuring endpoint permissions (Authentication vs Authorization)
- 🐥 Design a data model from scratch based on user requirements

**Comment by minjupgeorge** [comment]
##  Week 1
Sorry for the late check-in comment. @danielhalasz 
I was busy with my dance practice for our cultural program on september 17(onam 2022).

## I Need Help With:
Since I didn't attend the class in person ,first I went trough the class recordings twice and I understood the concept except  the explanations on white board.

while doing exercises I got stuck with filter concept used at Postman, I think i have to complete the exercises to understand them clearly

## What went well?
Following topics went well
 - Content Management system 
 -  API with path parameters and query parameters. 
 - Installed Postman API testing tool and Strapi 4

## Sunday Prep Work
Just started doing the exercises ,yet to complete more.

**Comment by danielhalasz** [comment]
> while doing exercises I got stuck with filter concept used at Postman, I think i have to complete the exercises to understand them clearly

yes I think so, once you do a few queries yourself, it will be clear, if not, let me know

**Comment by minjupgeorge** [comment]
##  Week 1

This week is better when compared to last week . I was able to spend some time for Strapi and its really interesting

## I Need Help With:
Exercises are really challenging . I understood sort ,filter concept now am trying to do queries with populate. its taking some time.
This week assignment went well. I have created chat application in Strapi and able to try API calls via postman. I have just started to get the channel details from strapi to my HTML page. still lot of confusions are there. I need time to complete it.

## What went well?
chat application via Strapi . Now I have clearly understood how to create JSON db for a front end webpage.
![image](https://user-images.githubusercontent.com/32439811/190410727-2adc589c-70d7-4ddf-b60e-4cc2f1272619.png)


## Sunday Prep Work
I am trying to fetch channel data from my strapi to index.html.
I wish to complete exercises before sunday.I

**Comment by minjupgeorge** [comment]
##  Week 3

## I Need Help With:
so far I have retrieved the data from chat application and now working with authentication. My work is [here](https://github.com/minjupgeorge/ChatApplication)

Is it possible to send message data from front end to strapi ?

 Now I have retrieved data from strapi,for me it took some time to retrieve it. Class help comments of @danielhalasz , @yoshimalaise ,@maria and @cristobal-lopez were very useful.
Thanks alot @yoshimalaise . Classes was so encouraging to build an application using strapi with java script especially the demo of wok restaurant was an eye opener for me to start coding again. 

## What went well?
Following topics went well
 
 -  API with path parameters and query parameters. 
 - Retrieving data from strapi to HTML

## Sunday Prep Work
Trying to finish Authentication
Starting watching videos of React which pointed by Kevin at slack.

**Comment by minjupgeorge** [comment]
RETROSPECTIVE 
## Me

### What was not clear, where did you get stuck?
Most of the time i got confused with attribute names to retrieve data nd dom model to display data .

### What was clear, what did you master?
I understood to create database using strapi especially relationships and able to read data from it

### Where can you still use some help?
I think i have to learn more about writing javascript functions and DOM manipulation

### Where can you help others moving forwards?
Past two modules I was not that confident to share my knowledge but still I used to discuss with my classmates to complete the project.

## The Course

### What can there be more of?
Nothing much. 

### What can there be less of?

### What material were most helpful (from HYF or elsewhere)?
Almost all resources that suggested by HYF is quite helpful.

### What HYF material was least helpful?

### Any suggestions for future classes?
More demo classes like wok-restaurant model.

**Comment by danielhalasz** [comment]
> Is it possible to send message data from front end to strapi ?

yes, with a POST request

**Comment by minjupgeorge** [comment]
I have a doubt @danielhalasz /  @yoshimalaise 
![image](https://user-images.githubusercontent.com/32439811/192832409-d3a4cfa7-fe26-434d-89a9-831b0fc01f90.png)

I can add new users to the 'User' content type which is default available @strapi  . How can I add new users to my chatuser content type

**Comment by danielhalasz** [comment]
well the exact same way I think, with a POST request..might have to set the Role permissions in the Strapi settings
